library/ssl_tls.c

```diff
@@ -4685,9 +4708,18 @@ int mbedtls_ssl_session_reset_int( mbedtls_ssl_context *ssl, int partial )
 #endif /* MBEDTLS_SSL_USE_MPS */
 
 #if defined(MBEDTLS_ZERO_RTT)
-    ssl->early_data_enabled = MBEDTLS_SSL_EARLY_DATA_DISABLED;
+#if defined(MBEDTLS_SSL_SRV_C)
+    if( ssl->early_data_server_buf != NULL &&
+        ssl->conf->max_early_data > 0 )
+    {
+        memset( ssl->early_data_server_buf, 0,  ssl->conf->max_early_data );
```

Comment:
What is the purpose of this? 

If we're about to run another connection, I think we can just overwrite the old 0-RTT contents without clearing them first.

If we're about to free the SSL context, we have to use reliable zeroization, that is, `mbedtls_platform_zeroize()`, as indicated above.

Reply:
OK, let's remove this.
I was following the handling of `in_buf` and `out_buf`.